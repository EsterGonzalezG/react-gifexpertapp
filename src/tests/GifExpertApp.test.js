import { shallow } from 'enzyme';
import React from 'react';
import { GifExpertApp } from './../GifExpertApp';

describe('test GifExpertApp', () => {
  const wrapper = shallow(<GifExpertApp />);

  test('should be render component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should be show list of categories', () => {
    const categories = ['Hello', 'Bye'];
    const wrapper = shallow(<GifExpertApp defaultCategoties={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });
});
