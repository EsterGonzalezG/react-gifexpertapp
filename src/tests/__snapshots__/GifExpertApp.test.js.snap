// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test GifExpertApp should be render component 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
</Fragment>
`;

exports[`test GifExpertApp should be show list of categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="Hello"
      key="Hello"
    />
    <GifGrid
      category="Bye"
      key="Bye"
    />
  </ol>
</Fragment>
`;
